### DISCUSSION TITLE
All services running but sensor is showing in Grid with status Fault

### DISCUSSION THREAD

**Discussion by ranguli** [Security-Onion-Solutions/securityonion#5140] [Unsupported Versions] [answered]
Hello,

I've deployed a sensor using the Security Onion AMI, and the install was a success. However, the sensor is showing up with a status of "Fault" on the Grid page of the SOC. It is reporting no events.

<img width="1452" alt="Screen Shot 2021-08-12 at 2 48 53 PM" src="https://user-images.githubusercontent.com/5544782/129240341-1ed3fe37-18c3-4b08-ac0f-2b19b685e1ce.png">

The end of `salt-call state.highstate` returned no failures:
```
Summary for local
--------------
Succeeded: 238 (changed=13)
Failed:      0
--------------
Total states run:     238
Total run time:    55.361 s
```

And `so-status` returned all services functioning normally:

```

Checking Docker status

    Docker --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]

Checking container statuses

    so-filebeat ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-nginx ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-sensoroni --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-steno ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-strelka-backend --------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-strelka-coordinator ----------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-strelka-filestream ------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-strelka-frontend -------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-strelka-gatekeeper ------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]
    so-strelka-manager --------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-suricata ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-telegraf ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-wazuh ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
    so-zeek -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]
```

I tried running `so-test` on the sensor, and the EPS increases (on the manager, not the sensor) and the command finishes successfully but I don't see any results in the SOC or other signs of life indicating that things "worked". 

Any help on the subject would be greatly appreciated, thank you! I'm expecting to be able to see network traffic in the environment that the sensor is deployed in but there may be some additional configuration required that I am missing.

**Comment by jertel** [comment, marked as answer]
When you expand the sensor node on the grid page, it should tell you which area is faulting: Process, Connection, Raid. Raid will usually say unknown unless you're running a Security Onion hardware appliance.

If Connection is faulted then you will need to investigate why the sensor isn't able to check-in to the manager node. Look at `/opt/so/log/sensoroni/sensoroni.log`.

If Process is faulting then you will need to look at the sensor's `/opt/so/log/status/status.log` to see if Telegram is detecting a fault in the sensor processes. There are other discussions here with more troubleshooting steps for this type of issue.

**Reply by ranguli** [reply]
Both Process and RAID are faulting. There was no file at `/opt/so/log/status/status.log`, but I did find `/opt/so/log/sostatus/status.log`, however it was empty. I will try updating from 2.6.0 -> 2.7.0 and hope it fixes the issue as @charlespctech mentioned.

**Reply by TOoSmOotH** [reply]
@ranguli If raid is faulting that means you are running a SO appliance. If that is the case please contact support and they can assist.

**Reply by rwwiv** [reply]
@ranguli We've discovered this is actually a bug and have created an issue to track the fix (see #5449)

**Reply by TOoSmOotH** [reply]
@ranguli You should be able to make the changes to /opt/so/saltstack/default/salt/common/tools/sbin/so-raid-status

```
  {%- if grains['sosmodel'] in ['SO2AMI01', 'SO2GCI01', 'SO2AZI01'] %}
  exit 0
  {%- endif %}
```

https://github.com/Security-Onion-Solutions/securityonion/pull/5450/files

**Reply by ranguli** [reply]
Thanks so much folks, I appreciate it! The deployment is not running on an SO appliance. I will make those changes and report back if there are any issues.
